test/file/main.c

```diff
@@ -203,6 +215,236 @@ static int __test_posix_fallocate(const char *file_path) {
     return 0;
 }
 
+#ifndef __GLIBC__
+#define FALLOC_FL_COLLAPSE_RANGE (0x08)
+#define FALLOC_FL_ZERO_RANGE (0x10)
+#define FALLOC_FL_INSERT_RANGE (0x20)
+#define FALLOC_FL_UNSHARE_RANGE (0x40)
+#endif
+
+static int __test_fallocate_with_invalid_mode(const char *file_path) {
+    int fd = open(file_path, O_RDWR);
+    if (fd < 0) {
+        THROW_ERROR("failed to open a file to read/write");
+    }
+    off_t len = 2 * BLK_SIZE;
+    if (fill_file_with_repeated_bytes(fd, len, 0xFF) < 0) {
+        THROW_ERROR("failed to fill file");
+    }
+    int mode = FALLOC_FL_KEEP_SIZE | 0xDEAD;
+    off_t offset = 0;
+    len = len / 2;
+    int ret = fallocate(fd, mode, offset, len);
+    if (ret >= 0 || errno != EOPNOTSUPP) {
```

Comment:
I think `ret < 0 && errno == EOPNOTSUPP` is more straightforward.

Reply:
`!(ret < 0 && errno == EOPNOTSUPP)` is equal to `ret >= 0 || errno != EOPNOTSUPP`